core/src/main/java/org/apache/iceberg/util/PartitionSet.java

```diff
@@ -200,7 +200,7 @@ public String toString() {
           StringBuilder partitionStringBuilder = new StringBuilder();
           partitionStringBuilder.append(structType.fields().get(i).name());
           partitionStringBuilder.append("=");
-          partitionStringBuilder.append(s.get(i, Object.class).toString());
+          partitionStringBuilder.append(String.valueOf(s.get(i, Object.class)));
```

Comment:
```suggestion
          partitionStringBuilder.append(s.get(i, Object.class));
```

The `String.valueOf` call isn't really necessary here

Reply:
I verified, and that's true. Thanks @nastra